packages/grafana-e2e/src/support/url.ts

```diff
@@ -12,3 +12,60 @@ export const getDashboardUid = (url: string): string => {
     return matches[1];
   }
 };
+
+/**
+ * Parses an escaped url query string into key-value pairs.
+ * Attribution: Code dervived from https://github.com/angular/angular.js/master/src/Angular.js#L1396
+ * @returns {Object.<string,boolean|Array>}
+ */
+export function parseKeyValue(keyValue: string) {
```

Comment:
I ended up just copying this utility over from grafana-data - for one, cypress really didn't like us refering to that package (and it honestly didn't make sense to refer to other packages in our test definitions) and for two, it didn't seem to make sense to add in links to the entire grafana-data package for one utility function. This does make some redundant code, but it keeps everything separated more which will probably be good. I can see us continuing to want to test that actions in the UI affect the URL as expected as well.

Reply:
Makes sense 👍 I wonder if we could avoid duplication by using URLSearchParams, see comment here https://github.com/grafana/grafana/pull/45020#discussion_r804598362. Though this might be a recurring problem if there are utilities that are not Grafana specific but still useful in testing.